appholder/src/main/java/com/android/mdl/app/fragment/TransferDocumentFragment.kt

```diff
@@ -246,7 +242,26 @@ class TransferDocumentFragment : Fragment() {
     private fun authenticationSucceeded() {
         // Send response again after user biometric authentication
         try {
-            vm.sendResponse()
+            if (vm.sendResponse() != null) {
+                // If this returns non-null it means that user-auth is still required.
+                //
+                // Wait, what? We just authenticated, so how can this happen?
+                //
+                // The answer is that it can happen when using face authentication b/c the
+                // countdown to when the key is no longer considered unlocked starts when the
+                // face is authenticated, NOT when the user presses the "Confirm" button. So
```

Comment:
It feels like this is an indication we're doing the wrong thing on "confirm". That is, instead of waiting for confirm to use the key, we should go ahead and do the crypto operation. Then, confirm can gate whether we actually send the data.

Reply:
Well, "Confirm" is part of the `BiometricPrompt` implementation, there's really nothing we can do about this.